numpy/random/mtrand/mtrand.pyx

```diff
@@ -158,7 +158,9 @@ cdef object cont0_array(rk_state *state, rk_cont0 func, object size,
     cdef npy_intp i
 
     if size is None:
-        return func(state)
+        with lock, nogil:
+            rv = func(state)
```

Comment:
That's what I wrote at first, but Cython forbids returning without the GIL, apparently. (Seems like a bug in Cython, but whatever).